src/colmap/estimators/cost_functions.h

```diff
@@ -84,16 +90,19 @@ class ReprojErrorCostFunction {
                             point3D_in_cam[0],
                             point3D_in_cam[1],
                             point3D_in_cam[2],
-                            &residuals[0],
-                            &residuals[1]);
-    residuals[0] -= T(observed_x_);
-    residuals[1] -= T(observed_y_);
+                            &residuals_ptr[0],
+                            &residuals_ptr[1]);
+    residuals_ptr[0] -= T(observed_x_);
+    residuals_ptr[1] -= T(observed_y_);
+    Eigen::Map<Eigen::Matrix<T, 2, 1>> residuals(residuals_ptr);
```

Comment:
What's the performance impact of this change?

Reply:
+1, we might need to benchmark it on a large BA problem.

Reply:
Made a new update here where the cost adaptively decides how to use the given covariance (identity / diagonal / general). With this update I believe we can officially drop all the IsotropicCovariance implementation. We could switch to the ``WeightedCostFunction`` when ceres has this supported. 

Will post the performance benchmarking on a large BA problem with the wheels built in this PR when it is available (i found that the wheels are much faster than the build from source in terms of BA performance)